r/R/dplyr-summarize.R

```diff
@@ -218,8 +218,16 @@ summarize_eval <- function(name, quosure, ctx, hash, recurse = FALSE) {
   } else if (all(inner_agg_exprs)) {
     # Something like: fun(agg(x), agg(y))
     # So based on the aggregations that have been extracted, mutate after
+    agg_field_refs <- make_field_refs(names(ctx$aggregations))
+    agg_field_types <- lapply(ctx$aggregations, function(x) x$data$type())
```

Comment:
`ctx$aggregations` are (a list of) expressions at this point, yeah?

Reply:
Ah, ok that's helpful. I was a bit hopeful that it was one of our R6 objects that might already have (or we could easily extend to have) a `$type()` method, but it looks like not.